sdk/react/src/use_picovoice.ts

```diff
@@ -9,230 +9,178 @@
   specific language governing permissions and limitations under the License.
 */
 
-import { useState, useEffect, useRef } from 'react';
+import { useState, useRef, useCallback, useEffect } from 'react';
 
 import { WebVoiceProcessor } from '@picovoice/web-voice-processor';
 
 import {
+  PicovoiceOptions,
   PicovoiceWorker,
-  PicovoiceWorkerFactory,
-  PicovoiceWorkerResponse,
-} from '@picovoice/picovoice-web-core';
-
-import { PorcupineKeyword } from '@picovoice/porcupine-web-core';
-
-import { RhinoContext, RhinoInference } from '@picovoice/rhino-web-core';
-
-export type PicovoiceHookArgs = {
-  accessKey: string;
-  porcupineKeyword: PorcupineKeyword;
-  rhinoContext: RhinoContext;
-  endpointDurationSec?: number;
-  requireEndpoint?: boolean;
-  start?: boolean;
-};
-
-type EngineControlType = 'ppn' | 'rhn';
-
-export function usePicovoice(
-  picovoiceWorkerFactory: PicovoiceWorkerFactory | null,
-  picovoiceHookArgs: PicovoiceHookArgs | null,
-  keywordCallback: (keywordLabel: string) => void,
-  inferenceCallback: (inference: RhinoInference) => void
-): {
+  PorcupineDetection,
+  PorcupineKeyword,
+  PorcupineModel,
+  RhinoContext,
+  RhinoInference,
+  RhinoModel,
+} from '@picovoice/picovoice-web';
+
+export function usePicovoice(): {
+  wakeWordDetection: PorcupineDetection | null;
+  inference: RhinoInference | null;
   contextInfo: string | null;
   isLoaded: boolean;
   isListening: boolean;
-  isError: boolean | null;
-  errorMessage: string | null;
-  engine: EngineControlType;
-  webVoiceProcessor: WebVoiceProcessor | null;
-  start: () => void;
-  pause: () => void;
-  stop: () => void;
+  error: string | null;
+  init: (
+    accessKey: string,
+    keyword: PorcupineKeyword,
+    porcupineModel: PorcupineModel,
+    context: RhinoContext,
+    rhinoModel: RhinoModel,
+    options?: PicovoiceOptions
+  ) => Promise<void>;
+  start: () => Promise<void>;
+  stop: () => Promise<void>;
+  release: () => Promise<void>;
 } {
```

Comment:
tslint and prettier don't agree on this it's very annoying
```suggestion
  } {
```